### PR TITLE
Added support for Winget packages

### PR DESCRIPTION
**Pull request by CodeCyclone**
Installing winget packages (via resource) as an option for cupboard.

Documentation - https://docs.microsoft.com/en-us/windows/package-manager/

Implementation Detail:
Winget list will report any installed application on the computer and `winget list -s winget` has a bug where it doesn't filter to only packages with known winget packages. Code will query the tool with the desired package name to filter out noise, otherwise you might get false positives (example: specify "code" and it would accidently match "Visual Studio Code").

### CODE REVIEW

File: src/Cupboard.Tests/Unit/Resources/WingetProviderTests.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            [WindowsFact]
+            public void Should_Uninstall_Package_If_Present()
+            {
+                // Given
+                var fixture = new CupboardFixture();
+                fixture.Security.IsAdmin = true;
+                fixture.Configure(ctx => ctx.UseManifest<Manifests.Uninstall>());
+
+                fixture.Process.Register("winget", "list --source winget --id GitHub.cli",
+                    new ProcessRunnerResult(0, "GitHub CLI GitHub.cli 1.12.1"));
```
Comment (patriksvensson) [inline src/Cupboard.Tests/Unit/Resources/WingetProviderTests.cs:93]: I suspect this needs to be something like this for the test to pass.

```suggestion
                fixture.Process.Register("winget", "list --source winget --id GitHub.cli",
	    		      new ProcessRunnerResult(0, "GitHub CLI GitHub.cli 1.12.1"),
    	    		  new ProcessRunnerResult(int.MinValue, "No installed package found matching input criteria."));
```

**Review by patriksvensson — CHANGES_REQUESTED** [review]
@CodeCyclone Awesome contribution! Much appreciated. 🎉 

A test is failing which I think happens because of a missing registration. I posted some code in the `WingetProviderTests.cs` file which I think should solve this.

File: src/Cupboard.Tests/Unit/Resources/WingetProviderTests.cs (hunk 1 above)
Comment (CodeCyclone) [reply src/Cupboard.Tests/Unit/Resources/WingetProviderTests.cs]: Good catch! For some reason my IDE wasn't catching it when I executed the tests

**Comment by patriksvensson** [comment]
Merged! Thank you for your contribution. Much appreciated! 👍

**Pull request merged**